pkg/daemon/daemon.go

```diff
@@ -107,7 +107,14 @@ func (dn *Daemon) Run(stop <-chan struct{}) error {
 		glog.Errorf("Marking degraded due to: %v", err)
 		return setUpdateDegraded(dn.kubeClient.CoreV1().Nodes(), dn.name)
 	} else if state == MachineConfigDaemonStateDegraded {
-		return fmt.Errorf("Node is degraded; exiting loudly...")
+		// just sleep so that we don't clobber output of previous run which
+		// probably contains the real reason why we marked the node as degraded
+		// in the first place
+		glog.Info("Node is degraded; going to sleep")
+		select {
```

Comment:
This looks good to me. Cleaner than my `for{}` loops for sure. The path here then would be:

- If the machine was already degraded, hold (unless explicitly told to stop via the channel)
- If we get an error during `getNodeAnnotation`, set degraded, exit, and hit the above line on reschedule and hold
- Otherwise, continue into main loop

Reply:
Yup, that's correct!